metrics/cmd/heapster-apiserver/app/server.go

```diff
@@ -60,12 +61,11 @@ func NewHeapsterApiServer(s *options.HeapsterRunOptions, metricSink *metricsink.
 }
 
 func newAPIServer(s *options.HeapsterRunOptions) (*genericapiserver.GenericAPIServer, error) {
-	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("heapster.kube-system"); err != nil {
+	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
```

Comment:
I followed this code https://github.com/kubernetes/kubernetes/blob/017e61c76f885158aa8cc0dfa53ab6a1db0b1380/staging/src/k8s.io/kube-apiextensions-server/pkg/cmd/server/start.go#L90. I'll try to change it and see whether it works.